### PR TITLE
DEV: Allow TopicEmbed.import to optionally receive a list of tags

### PR DESCRIPTION
**Pull request by xfalcox**
This will be used by the rss-polling plugin

### CODE REVIEW

File: app/models/topic_embed.rb [hunk 1]
```diff
@@ -58,7 +58,8 @@ def self.import(user, url, title, contents, category_id: nil, cook_method: nil)
           raw: absolutize_urls(url, contents),
           skip_validations: true,
           cook_method: cook_method,
-          category: category_id || eh.try(:category_id)
+          category: category_id || eh.try(:category_id),
+          tags: tags,
```
Comment (tgxworld) [inline app/models/topic_embed.rb:62]: If tagging is disabled, we should exclude the tags from `create_args`

File: app/models/topic_embed.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@ def self.imported_from_html(url)
   end
 
   # Import an article from a source (RSS/Atom/Other)
-  def self.import(user, url, title, contents, category_id: nil, cook_method: nil)
+  def self.import(user, url, title, contents, category_id: nil, cook_method: nil, tags: nil)
```
Comment (tgxworld) [inline app/models/topic_embed.rb:30]: If we're planning to use this from a plugin, can we expose this method via the plugin API instead? That way, we'll at least be able to better manager deprecation on this method in the future.

File: app/models/topic_embed.rb (hunk 1 above)
Comment (xfalcox) [reply app/models/topic_embed.rb]: Handled in 8288985

File: app/models/topic_embed.rb (hunk 2 above)
Comment (xfalcox) [reply app/models/topic_embed.rb:30]: IMO is generally useful being able to add a category and tags to an imported topic in the codebase.

Comment (tgxworld) [reply app/models/topic_embed.rb:30]: It may be useful but there isn’t anywhere in core that is using this right now.

Comment (xfalcox) [reply app/models/topic_embed.rb:30]: Can you expand a bit on this?

Do you want me to create a copy of this method under `/lib/plugin/instance.rb` but with the extra flag parameter and call that new version from 

https://github.com/discourse/discourse-rss-polling/blob/3339e5983fd13f95167ef143283f0026b892171c/app/jobs/jobs/discourse_rss_polling/poll_feed.rb#L38

?

Comment (eviltrout) [reply app/models/topic_embed.rb:30]: I don't think we should hold up this PR any longer. Please merge. @tgxworld can you follow up on dev with a post about what you are looking for?

**Review by eviltrout — APPROVED** [review]

**Pull request merged**